src/commands/command_utils.ts

```diff
@@ -0,0 +1,93 @@
+import { Uri, Diagnostic, Range, DiagnosticSeverity } from "vscode";
+import { NodeMetaData } from "../domain";
+import { DBTProject } from "../manifest/dbtProject";
+
+export function findModelProblems(
+  projectModel: NodeMetaData,
+  modelDict: any[],
+  name: string,
+  project: DBTProject,
+  allDiagnostic: { [fullFilePath: string]: Diagnostic[] },
+) {
+  const existingColumnsLowered = Object.keys(projectModel.columns).map((key) =>
+    key.toLowerCase(),
+  );
+  const allDBColumns = modelDict.map(({ column_name }) =>
+    column_name.toLowerCase(),
+  );
+  if (!projectModel.patch_path) {
+    // TODO - file an error if the model is not documented at all (no doc entry exists)
+    const err_message = "No documentation for model: " + name;
+    console.log(err_message);
+    let projDiagnostic =
+      allDiagnostic[
+        Uri.joinPath(project.projectRoot, DBTProject.DBT_PROJECT_FILE).fsPath
+      ];
+    if (projDiagnostic === undefined) {
+      allDiagnostic[
+        Uri.joinPath(project.projectRoot, DBTProject.DBT_PROJECT_FILE).fsPath
+      ] = projDiagnostic = [];
+    }
+    projDiagnostic.push(
+      new Diagnostic(
+        new Range(0, 0, 0, 0),
+        err_message,
+        DiagnosticSeverity.Information,
+      ),
+    );
+  }
+  for (const column of modelDict) {
+    if (!existingColumnsLowered.includes(column.column_name.toLowerCase())) {
+      const err_message =
+        "Column undocumented for model: " +
+        name +
+        "\n" +
+        "Expected an entry for: " +
+        column.column_name;
+
+      // TODO - this should point to the model file ?
```

Comment:
Please point this indeed to the model file. Should be doable based on the manifest.